src/js/components/RefileWarning.jsx

```diff
@@ -0,0 +1,42 @@
+import React, { PropTypes } from 'react'
+import { Link } from 'react-router'
+import { parseLocation } from '../api'
+
+const refileText = 'Syntactical and validity edits require file resubmission.'
+const validateText = 'Quality and macro edits must be validated before continuing.'
+
+const getText = (props) => {
+  let textToRender = null
+  let refileLink = null
+
+  if(props.types.syntactical.edits.length !== 0 ||
+     props.types.validity.edits.length !== 0){
+       textToRender = refileText
+       refileLink = getRefileLink(props)
+     }else{
+       textToRender = validateText
+     }
+
+  return <h3><span className="cf-icon cf-icon-error cf-icon__3x"></span><span className="refile-text">{textToRender}{refileLink?' ':''}{refileLink}</span></h3>
+}
+
+const getRefileLink = (props) => {
+  const location = parseLocation()
+  const href = `/${location.id}/${location.filing}`
```

Comment:
Couldn't we just use `props.params` here? Then we could remove the `import { parseLocation } from '../api'` and not use `parseLocation()`. 

It could just be `return <Link to={'/${props.params.institution}/${props.params.filing}'} onClick={props.refileLink}>Refile here.</Link>`